File: python-d_and_algo/searchAlgo/binarySearch.py
import math
def binarySearch(sortedArray, item):
	length = len(sortedArray)
	start = 0
	last = length-1
	middle = math.floor((start + last)/2)

	for i in range(length):
		if sortedArray[middle] != item and start <= last:
			if item < sortedArray[middle]:
				last = middle-1
			elif item > sortedArray[middle]:
				start = middle + 1

			middle = math.floor((start + last)/2)

		if sortedArray[middle] == item:
			return middle

	return -1

print(binarySearch([1, 2, 4, 7, 9, 10, 12, 14, 16, 18, 22, 24], 22))

File: searchingAlgo/binarySearch.js
const binarySearch = (sortedArr, item) => {
  if (!sortedArr || !item) return;
  const len = sortedArr.length;
  let first = 0,
    last = len - 1,
    middle = Math.floor((first + last) / 2);

  for (let i = 0; i < len; i++) {
    if (sortedArr[middle] !== item && first <= last) {
      if (item > sortedArr[middle]) {
        first = middle + 1;
      } else if (item < sortedArr[middle]) {
        last = middle - 1;
      }
      middle = Math.floor((first + last) / 2);
    }
    if (sortedArr[middle] === item) return middle;
  }
  return -1;
};
console.log(binarySearch([1, 2, 4, 7, 9, 10, 12, 14, 16, 18, 22, 24], 22));
